packages/permission-controller/src/Caveat.test.ts

```diff
@@ -9,7 +9,6 @@ describe('decorateWithCaveats', () => {
     const caveatSpecifications = {
       reverse: {
         type: 'reverse',
-        // TODO: Replace `any` with type
```

Comment:
Why can't this be typed to `Caveat`?

Reply:
Agreed, these `any`s look fixable. 

The "Replace `any`" TODOs shouldn't be removed just because they occur in test files. Most of our current `any`s in tests are not examples of intentional/purposeful usage that should be preserved ([ex1](https://github.com/MetaMask/contributor-docs/blob/240206-typescript-guidelines-any-as/docs/typescript.md#use-ts-expect-error-to-force-runtime-execution-of-a-branch-for-validation-or-testing), [ex2](https://github.com/MetaMask/contributor-docs/blob/240206-typescript-guidelines-any-as/docs/typescript.md#ts-expect-error-may-be-acceptable-to-use-in-tests-to-intentionally-break-features)). It looks like this also applies to the test files in this PR, although I'm taking a closer look to see if there are any exceptions.

Example fixes for this file:

```diff
diff --git a/packages/permission-controller/src/Caveat.test.ts b/packages/permission-controller/src/Caveat.test.ts
index 6a9dc6cc..697ab499 100644
--- a/packages/permission-controller/src/Caveat.test.ts
+++ b/packages/permission-controller/src/Caveat.test.ts
@@ -1,4 +1,4 @@
-import type { PermissionConstraint } from '.';
+import type { Caveat, PermissionConstraint } from '.';
 import { decorateWithCaveats, PermissionType } from '.';
 import * as errors from './errors';
 
@@ -9,10 +9,11 @@ describe('decorateWithCaveats', () => {
     const caveatSpecifications = {
       reverse: {
         type: 'reverse',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        decorator: (method: any, _caveat: any) => async () => {
-          return (await method()).reverse();
-        },
+        decorator:
+          (method: () => Promise<unknown[]>, _caveat: Caveat<string, null>) =>
+          async () => {
+            return (await method()).reverse();
+          },
       },
     };
 
@@ -45,17 +46,19 @@ describe('decorateWithCaveats', () => {
     const caveatSpecifications = {
       reverse: {
         type: 'reverse',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        decorator: (method: any, _caveat: any) => async () => {
-          return (await method()).reverse();
-        },
+        decorator:
+          (method: () => Promise<unknown[]>, _caveat: Caveat<string, null>) =>
+          async () => {
+            return (await method()).reverse();
+          },
       },
       slice: {
         type: 'slice',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        decorator: (method: any, caveat: any) => async () => {
-          return (await method()).slice(0, caveat.value);
-        },
+        decorator:
+          (method: () => Promise<unknown[]>, caveat: Caveat<string, number>) =>
+          async () => {
+            return (await method()).slice(0, caveat.value);
+          },
       },
     };
 
@@ -111,10 +114,11 @@ describe('decorateWithCaveats', () => {
     const caveatSpecifications = {
       reverse: {
         type: 'reverse',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        decorator: (method: any, _caveat: any) => async () => {
-          return (await method()).reverse();
-        },
+        decorator:
+          (method: () => Promise<unknown[]>, _caveat: Caveat<string, null>) =>
+          async () => {
+            return (await method()).reverse();
+          },
       },
     };
```